File: src/2022/day05/src/test/java/ca/terrylockett/aoc2022/day05/TestDay05.java
package ca.terrylockett.aoc2022.day05;

import ca.terrylockett.aoccommon.resources.Resources;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

import java.net.URISyntaxException;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class TestDay05 {

	static final String TEST_FILE_NAME = "test_input.txt";
	static String testFilePath = "";

	@BeforeAll
	public static void setup() throws URISyntaxException {
		testFilePath = Resources.getInputFilePath(TEST_FILE_NAME).orElseThrow();
	}

	@Test
	public void day5Part1() throws Exception {
		String result = SupplyStacks.unload(testFilePath);
		assertEquals("CMZ", result);
	}

	@Test
	public void day5Part2() throws Exception {
		String result = SupplyStacks.unload(testFilePath, true);
		assertEquals("MCD", result);
	}
}

File: src/2023/day05/src/test/kotlin/ca/terrylockett/aoc2023/day05/TestDay05.kt
package ca.terrylockett.aoc2023.day05;

import ca.terrylockett.aoccommon.inputfilefinder.InputFileFinder;
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.TestInstance

@TestInstance(TestInstance.Lifecycle.PER_CLASS)
class TestDay05 {

   companion object {
        const val TEST_FILE_NAME = "test-input.txt"
    }

    @Test
    fun part1() {
        val inputFilePath = InputFileFinder.getInputFilePath(TEST_FILE_NAME).orElseThrow()

        assertEquals(35L, part1(inputFilePath))
    }
    
//    @Test
//    fun part2() {
//        val inputFilePath = InputFileFinder.getInputFilePath(TEST_FILE_NAME).orElseThrow()
//        val engine = EngineSchematic(inputFilePath)
//
//        assertEquals(467835, engine.part2FindGears())
//    }
}
